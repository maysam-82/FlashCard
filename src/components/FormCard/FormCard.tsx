import React from 'react';
import { FormControl, FormGroup, FormLabel } from 'react-bootstrap';
import { ICard } from '../../types/card';

export interface IProps {
	id: number;
	prompt: string;
	answer: string;
	updateCard: (
		event: React.ChangeEvent<HTMLInputElement>,
		id: number,
		key: keyof ICard
	) => void;
}

const FormCard: React.SFC<IProps> = ({ id, updateCard, prompt, answer }) => {
	return (
		<FormGroup className="border-bottom p-2 mb-2 d-flex align-items-center">
			<FormLabel className="m-0">Prompt:</FormLabel>
			<FormControl
				onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
					updateCard(event, id, 'prompt')
				}
				value={prompt}
				className="ml-2 mr-4"
			/>
			<FormLabel className="m-0 ml-2">Answer:</FormLabel>
			<FormControl
				onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
					updateCard(event, id, 'answer')
				}
				value={answer}
				className="ml-2"
			/>
		</FormGroup>
	);
};

export default FormCard;
