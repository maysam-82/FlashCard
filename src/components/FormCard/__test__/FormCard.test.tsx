import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import FormCard, { IProps } from '../FormCard';

const props: IProps = {
	id: 600,
	prompt: 'sample p',
	answer: 'sample a',
	updateCard: jest.fn(),
};
describe('FormCard component', () => {
	let component: ShallowWrapper;
	beforeEach(() => {
		component = shallow(<FormCard {...props} />);
	});
	it('should render FormCard', () => {
		expect(component).toMatchSnapshot();
	});
	it('should update `prompt` input value equal to `sample p`', () => {
		expect(component.find('FormControl').at(0).props().value).toEqual(
			props.prompt
		);
	});
	it('should update `answer` input value equal to `sample a`', () => {
		expect(component.find('FormControl').at(1).props().value).toEqual(
			props.answer
		);
	});

	describe('When `prompt` input is changed ', () => {
		beforeEach(() => {
			component = shallow(<FormCard {...props} />);
		});
		// Clear all mock functions after each test
		afterEach(() => {
			jest.clearAllMocks();
		});
		it('should call `updateCard` in `prompt` field ', () => {
			component
				.find('FormControl')
				.at(0)
				.simulate('change', { target: { value: props.prompt } });
			expect(props.updateCard).toHaveBeenCalledTimes(1);
		});
	});
	describe('When `answer` input is changed ', () => {
		beforeEach(() => {
			component = shallow(<FormCard {...props} />);
		});
		// Clear all mock functions after each test
		afterEach(() => {
			jest.clearAllMocks();
		});
		it('should call `updateCard` in `answer` field ', () => {
			component
				.find('FormControl')
				.at(1)
				.simulate('change', { target: { value: props.answer } });
			expect(props.updateCard).toHaveBeenCalledTimes(1);
		});
	});
});
