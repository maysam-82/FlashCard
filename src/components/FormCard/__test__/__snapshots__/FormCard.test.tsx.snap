// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCard component should render FormCard 1`] = `
<FormGroup
  className="border-bottom p-2 mb-2 d-flex align-items-center"
>
  <FormLabel
    className="m-0"
    column={false}
    srOnly={false}
  >
    Prompt:
  </FormLabel>
  <FormControl
    className="ml-2 mr-4"
    onChange={[Function]}
    value="sample p"
  />
  <FormLabel
    className="m-0 ml-2"
    column={false}
    srOnly={false}
  >
    Answer:
  </FormLabel>
  <FormControl
    className="ml-2"
    onChange={[Function]}
    value="sample a"
  />
</FormGroup>
`;
